### PR TITLE
Wip/6.0 polymorphic set

### PR DESCRIPTION
**Pull request by dreab8**

### CODE REVIEW

File: hibernate-core/src/test/java/org/hibernate/orm/test/bytecode/enhancement/lazy/proxy/LazyToOnesProxyWithSubclassesTest.java [hunk 1]
```diff
@@ -281,8 +281,6 @@ public void cleanUpTestData() {
 		inTransaction(
 				session -> {
 					session.createQuery( "delete from OtherEntity" ).executeUpdate();
-					session.createQuery( "delete from Human" ).executeUpdate();
-					session.createQuery( "delete from Primate" ).executeUpdate();
```
Comment (beikov) [inline hibernate-core/src/test/java/org/hibernate/orm/test/bytecode/enhancement/lazy/proxy/LazyToOnesProxyWithSubclassesTest.java:285]: What is this for?

**Review by beikov — COMMENTED** [review]
The overall change looks ok to me but I'm having a hard time understanding why/when an initializer is called multiple times. Can you elaborate a bit on the case?

**Review by beikov — COMMENTED** [review]
The overall change looks ok to me but I'm having a hard time understanding why/when an initializer is called multiple times. Can you elaborate a bit on the case?

**Comment by dreab8** [comment]
The assembler can call the initializer, one example is https://github.com/hibernate/hibernate-orm/pull/4456/files#diff-c562e6899ef6d0ae3271173cc4dfc2c54df65de5b10737ce46c19e118c46eac3R258 and also I think for `EmbeddableId` with `toOne`associations.

File: hibernate-core/src/test/java/org/hibernate/orm/test/bytecode/enhancement/lazy/proxy/LazyToOnesProxyWithSubclassesTest.java (hunk 1 above)
Comment (dreab8) [reply hibernate-core/src/test/java/org/hibernate/orm/test/bytecode/enhancement/lazy/proxy/LazyToOnesProxyWithSubclassesTest.java:285]: I have an annoying failure when I run the tests locally, the issue with temporary tables during deletion, I made a tried and forget to revert this change for the PR

Comment (beikov) [reply hibernate-core/src/test/java/org/hibernate/orm/test/bytecode/enhancement/lazy/proxy/LazyToOnesProxyWithSubclassesTest.java:285]: I found the issue and fixed it as part of https://github.com/hibernate/hibernate-orm/pull/4425

**Pull request merged**